examples/cpp/gcp_observability/helloworld/greeter_client.cc

```diff
@@ -0,0 +1,115 @@
+#endif
+
+ABSL_FLAG(std::string, target, "localhost:50051", "Server address");
+
+using grpc::Channel;
+using grpc::ClientContext;
+using grpc::Status;
+using helloworld::Greeter;
+using helloworld::HelloReply;
+using helloworld::HelloRequest;
+
+class GreeterClient {
+ public:
+  GreeterClient(std::shared_ptr<Channel> channel)
+      : stub_(Greeter::NewStub(channel)) {}
+
+  // Assembles the client's payload, sends it and presents the response back
+  // from the server.
+  std::string SayHello(const std::string& user) {
+    // Data we are sending to the server.
+    HelloRequest request;
+    request.set_name(user);
+
+    // Container for the data we expect from the server.
+    HelloReply reply;
+
+    // Context for the client. It could be used to convey extra information to
+    // the server and/or tweak certain RPC behaviors.
+    ClientContext context;
+
+    // The actual RPC.
+    Status status = stub_->SayHello(&context, request, &reply);
+
+    // Act upon its status.
+    if (status.ok()) {
+      return reply.message();
+    } else {
+      std::cout << status.error_code() << ": " << status.error_message()
+                << std::endl;
+      return "RPC failed";
+    }
+  }
+
+ private:
+  std::unique_ptr<Greeter::Stub> stub_;
+};
+
+int main(int argc, char** argv) {
+  absl::ParseCommandLine(argc, argv);
+  // Instantiate the client. It requires a channel, out of which the actual RPCs
+  // are created. This channel models a connection to an endpoint specified by
+  // the argument "--target=" which is the only expected argument.
+  std::string target_str = absl::GetFlag(FLAGS_target);
+  // Turn on GCP Observability for the whole binary. Note that this should be
+  // done before any other gRPC operation.
+  auto status = grpc::experimental::GcpObservabilityInit();
+  if (!status.ok()) {
+    std::cerr << "GcpObservabilityInit() failed: " << status.ToString()
+              << std::endl;
+    return static_cast<int>(status.code());
+  }
+  std::cout << "Initialized GCP Observability" << std::endl;
+  // We indicate that the channel isn't authenticated (use of
+  // InsecureChannelCredentials()).
+  GreeterClient greeter(
+      grpc::CreateChannel(target_str, grpc::InsecureChannelCredentials()));
+  std::string user("world");
+  std::string reply = greeter.SayHello(user);
+  std::cout << "Greeter received: " << reply << std::endl;
+  // Flush out any pending Observability data
+  std::cout << "Closing GCP Observability" << std::endl;
+  grpc::experimental::GcpObservabilityClose();
+  std::cout << "Sleeping for 25 seconds to make sure Observability stats and "
+               "tracing are flushed. Don't shut off server either."
+            << std::endl;
+  // Currently, GcpObservabilityClose() only supports flushing logs. Stats and
+  // tracing get automatically flushed at a regular interval, so sleep for an
+  // interval to make sure that those are flushed too.
+  std::this_thread::sleep_for(std::chrono::seconds(25));
```

Comment:
Java decided to do add sleep in `close()`, which can be removed later once we have a fix so that shutdown would run faster.

Reply:
After talking with @fengli79, the decision was not to include the sleep in `close()`